parmed/openmm/parameters.py

```diff
@@ -137,7 +142,12 @@ def write(self, dest, provenance=None):
             own_handle = True
         else:
             own_handle = False
-        self.typeify_templates()
+        if self.atom_types:
+            try:
+                self.typeify_templates()
+            except:
```

Comment:
You should (almost) always avoid bare excepts, since it can mask real errors.  I think `KeyError` is the expected exception if things go wrong, yes?